File: functions/api/health.js
/**
 * EdgeOne Pages Node Function: Health Check
 * GET /api/health
 */
export default async function handler(request) {
  return new Response(
    JSON.stringify({
      status: 'ok',
      model: 'sora_video2'
    }),
    {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
        'Cache-Control': 'no-cache'
      }
    }
  );
}

File: api/health.py
"""
Vercel Serverless Function: Health Check
GET /api/health
"""
from http.server import BaseHTTPRequestHandler
import json


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Cache-Control', 'no-cache')
        self.end_headers()
        
        response = {
            'status': 'ok',
            'model': 'sora_video2'
        }
        
        self.wfile.write(json.dumps(response).encode())
        return

